test/optimizer/selectivity_util_test.cpp

```diff
@@ -0,0 +1,473 @@
+  const_value_expr_ptr =
+      std::make_unique<parser::ConstantValueExpression>(type::TypeId::INTEGER, execution::sql::Integer(0));
+
+  // Create a value condition to pass to SelectivityUtil.
+  value_condition = ValueCondition(catalog::col_oid_t(3), parser::ExpressionType::COMPARE_LESS_THAN_OR_EQUAL_TO,
+                                   std::move(const_value_expr_ptr));
+
+  res = selectivity_util.ComputeSelectivity(
+      common::ManagedPointer<NewColumnStats<execution::sql::Integer>>(&column_stats_obj_2_), value_condition);
+
+  // True value is 0.1. TODO(arvindsk) make '<=' better here.
+  ASSERT_DOUBLE_EQ(0, res);
+}
+
+// NOLINTNEXTLINE
+TEST_F(SelectivityUtilTests, TestIntegerGreaterThan) {
+  // TEST PART 1
+  // Create a constant value expression to pass to ValueCondition.
+  // Floating point numbers have DECIMAL type and are represented as Reals in execution layer.
+  std::unique_ptr<parser::ConstantValueExpression> const_value_expr_ptr =
+      std::make_unique<parser::ConstantValueExpression>(type::TypeId::INTEGER, execution::sql::Integer(6));
+
+  // Create a value condition to pass to SelectivityUtil.
+  ValueCondition value_condition(catalog::col_oid_t(3), parser::ExpressionType::COMPARE_GREATER_THAN,
+                                 std::move(const_value_expr_ptr));
+  // Compute selectivity for col > 6.
+  double res = selectivity_util.ComputeSelectivity(
+      common::ManagedPointer<NewColumnStats<execution::sql::Integer>>(&column_stats_obj_2_), value_condition);
+
+  // True value is 0.1. Acceptable error.
+  ASSERT_DOUBLE_EQ(0.11249999999999999, res);
+
+  // TEST PART 2
+  const_value_expr_ptr =
+      std::make_unique<parser::ConstantValueExpression>(type::TypeId::INTEGER, execution::sql::Integer(3));
+
+  // Create a value condition to pass to SelectivityUtil.
+  value_condition = ValueCondition(catalog::col_oid_t(3), parser::ExpressionType::COMPARE_GREATER_THAN,
+                                   std::move(const_value_expr_ptr));
+  // Compute selectivity for col > 3.
+  res = selectivity_util.ComputeSelectivity(
+      common::ManagedPointer<NewColumnStats<execution::sql::Integer>>(&column_stats_obj_2_), value_condition);
+
+  // True value is 0.5. TODO(arvindsk) Making '<=' better will fix this.
+  ASSERT_DOUBLE_EQ(0.6, res);
+
+  // TEST PART 3
+  const_value_expr_ptr =
+      std::make_unique<parser::ConstantValueExpression>(type::TypeId::INTEGER, execution::sql::Integer(0));
+
+  // Create a value condition to pass to SelectivityUtil.
+  value_condition = ValueCondition(catalog::col_oid_t(3), parser::ExpressionType::COMPARE_GREATER_THAN,
+                                   std::move(const_value_expr_ptr));
+  // Compute selectivity for col > 0.
+  res = selectivity_util.ComputeSelectivity(
+      common::ManagedPointer<NewColumnStats<execution::sql::Integer>>(&column_stats_obj_2_), value_condition);
+
+  // True value is 0.7. TODO(arvindsk) Making '<=' better will fix this.
+  ASSERT_FLOAT_EQ(0.8, res);
```

Comment:
Yes, the todo is a good one. It makes sense to check if the result is in the bounds of a particular value.